meps/mep-cross-references.md

```diff
@@ -0,0 +1,339 @@
+:::
+
+**Styling**
+
+All link syntax supports styling inside of the reference, (e.g. `[A **bolded _reference_** to a page](./myst.md)`) the reference role syntax currently does **not** support styling of the inner content.
+
+## Proposal
+
+We propose a cross-reference syntax that uses CommonMark links in all three forms. The goal is to support all use cases of cross-referencing with the most common use cases of referencing a document, file, section or element being simple, terse and familiar.
+
+**Overview:**
+
+| Existing Syntax                          | New Syntax                   |
+| :--------------------------------------- | :--------------------------- |
+| `[](my-id)`[^legacy_hash]                | `[](#my-id)`                 |
+| `` {ref}`my-id` ``                       | `[](#my-id)`                 |
+| `` {eq}`my-equation` ``                  | `[](#my-equation)`           |
+| `` {ref}`Custom Text <my-id>` ``         | `[Custom Text](#my-id)`      |
+| `` {ref}`See "{name}" <my-id>` ``        | `[See "%t"](#my-id)`         |
+| `` {numref}`Custom Number %s <my-id>` `` | `[Custom Number %s](#my-id)` |
+| `` {doc}`my-doc` ``                      | `[](my-doc.md)`              |
+| `` {doc}`my-doc` ``                      | `[](../examples/my-doc.md)`  |
+| `` {download}`my-doc.zip` ``             | `[](my-doc.zip)`             |
+
+[^legacy_hash]: This is backwards compatible, however, now raises a `xref_legacy` warning for old syntax.
+
+All of the above link examples can be easily complemented by both adding Title Link syntax and Reference Link syntax (i.e. `[Explicit *Markdown* text][label]`).
+We have omitted the auto-link syntax from the overview for brevity, they are shown in detail below.
+In all cases, the existing role syntax should continue to work and receive ongoing support from the parser(s).
+
+### Syntax
+
+The parts of the link are `[text](link "title")` with an optional scheme (`[text](scheme:link "title")`). The `"title"` is not modified in our proposed syntax. Auto Link syntax, `<scheme:link>`, requires the scheme to be present, we follow the CommonMark [definition of a scheme](https://spec.commonmark.org/0.30/#scheme).
+
+#### `text`
+
+If the `text` is not included, it will be filled in by the default of the target.
+
+- If the reference target is enumerated this will by default be "Sec 2.3"; "Fig. 1" for enumerated figures; "(5)" for equations; "Thm. 2", "Lemma 3", etc. depending on the target type. This can be customized in project configuration or frontmatter, however, the details of that are out of scope for this MEP.
+- If the reference target is not enumerated, the title will be used. This may be the section header text (including syntax/style) or a figure caption.
+- If the node is not enumerated and does not have a title, the reference label will be used.
+
+If the `text` is included it will be used as is with two additional template values (`%s` and `%t`)
+
+- Enumeration (`%s`)
```

Comment:
I don't think `%s` makes sense here.
The fact that sphinx uses it (as an old style format), is purely just to comply with https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting
But here, if we are going to use `%t` for title, then enumeration should similarly be represented by `%e`

Reply:
Maybe `%n`, I think that most people will think "number" rather than enumeration, even though that is what the spec is.

Reply:
Does this have to be a single letter. Could we use `%num`? I understand the idea behind `%e` (relative to `%t`) but I honestly don't think I would remember to us it.

FWIW - As a user I tend to make use of the `numref` role and accept the default `style` of numbered figures. I don't -- but a user could change the template for the `numref`  format and specify something like `Fig %s` rather than displaying `Figure %s` (for example) to change this document wide right?

Reply:
Yes! Regardless of the letter/sequence we use to interpolate the number into the text, it will be configurable at a project, page, and link/numref level.